library/std/src/f32.rs

```diff
@@ -978,7 +978,9 @@ impl f32 {
         unsafe { cmath::tgammaf(self) }
     }
 
-    /// Returns the natural logarithm of the gamma function.
+    /// Natural logarithm of the absolute value of the gamma function
+    ///
+    /// The integer part of the tuple indicates the sign of the gamma function.
```

Comment:
Would be okay to merge this as-is, but since it's worth explaining why this sign is useful in the first place, I decided to write up an explanation about how this can be used to compute the complex-valued logarithm from the returned value:

```suggestion
    /// Natural logarithm of the absolute value of the gamma function.
    ///
    /// Since the gamma function can be negative, its natural logarithm could
    /// be complex-valued and thus not normally representable. While the
    /// standard library doesn't offer any means to deal with complex values,
    /// this function still offers the real part of the resulting complex
    /// logarithm, which is equivalent to the logarithm of the absolute value
    /// of the input.
    ///
    /// To help construct the full complex value, the sign of the computed gamma
    /// function before taking the natural logarithm is offered as a second
    /// return value, which can trivially be converted into an imaginary part
    /// by multiplying by [`f32::PI`].
```

(Note that I say "an imaginary part" since the complex logarithm may add additional factors of 2pi, but that's not super important to say here. So, I am still remaining correct while not worrying anyone about it who doesn't really care.)

Reply:
Also worth mentioning if you do like this and want to copy it to the `f64` version, you should change the link to `f32::PI` to `f64::PI`.

This also is an indication of why returning the sign as a float instead of an int might be preferable, as discussed on the issue.

Reply:
I omitted the period on the initial line because it's not actually a complete sentence, nor does it need to be.

Reply:
I think "the sign [...] is offered as a second return value" needs explanation, i.e. -- this returns 1 on positive (or 0) gamma values and -1 on negative values.

Also while we are at it, do we want to change the sign to be represented as an `f32`? Otherwise it will require somewhat awkward explicit conversion back to a float type.